src/Microsoft.Identity.Web/AuthorizeForScopesAttribute.cs

```diff
@@ -99,6 +100,14 @@ public override void OnException(ExceptionContext context)
                         msalUiRequiredException,
                         context.HttpContext.User,
                         UserFlow);
+
+                    if (IsAjaxRequest(context.HttpContext.Request) && (!string.IsNullOrEmpty(context.HttpContext.Request.Headers["x-ReturnUrl"])
+                        || !string.IsNullOrEmpty(context.HttpContext.Request.Query["x-ReturnUrl"])))
```

Comment:
>x-ReturnUrl [](start = 84, length = 11)

nit: x-ReturnUrl can go into our contants file (also something we can do post-merge). :)

Reply:
Thanks, it should be.